internal/resources/product/resource.go

```diff
@@ -0,0 +1,574 @@
+				Description: "User-defined identifier used in a deep-link URL for the Product. It must be unique " +
+					"across a Project, but a Product can have the same slug in different Locales. It must match the " +
+					"pattern [a-zA-Z0-9_-]{2,256}.",
+				MarkdownDescription: "User-defined identifier used in a deep-link URL for the Product. It must be " +
+					"unique across a Project, but a Product can have the same slug in different " +
+					"[Locales](https://docs.commercetools.com/api/types#ctp:api:type:Locale). It must match the " +
+					"pattern `[a-zA-Z0-9_-]{2,256}`.",
+				Required: true,
+			},
+			"description": schema.MapAttribute{
+				CustomType:          customtypes.NewLocalizedStringType(),
+				Description:         "Description of the Product.",
+				MarkdownDescription: "Description of the Product.",
+				Optional:            true,
+			},
+			"categories": schema.ListAttribute{
+				ElementType:         types.StringType,
+				Description:         "Categories assigned to the Product.",
+				MarkdownDescription: "Categories assigned to the Product.",
+				Optional:            true,
+				Validators:          []validator.List{listvalidator.SizeAtLeast(1)},
+			},
+			"meta_title": schema.MapAttribute{
+				CustomType:          customtypes.NewLocalizedStringType(),
+				Description:         "Title of the Product displayed in search results.",
+				MarkdownDescription: "Title of the Product displayed in search results.",
+				Optional:            true,
+			},
+			"meta_description": schema.MapAttribute{
+				CustomType:          customtypes.NewLocalizedStringType(),
+				Description:         "Description of the Product displayed in search results.",
+				MarkdownDescription: "Description of the Product displayed in search results.",
+				Optional:            true,
+			},
+			"meta_keywords": schema.MapAttribute{
+				CustomType:          customtypes.NewLocalizedStringType(),
+				Description:         "Keywords that give additional information about the Product to search engines.",
+				MarkdownDescription: "Keywords that give additional information about the Product to search engines.",
+				Optional:            true,
+			},
+			"tax_category_id": schema.StringAttribute{
+				Description:         "The Tax Category to be assigned to the Product.",
+				MarkdownDescription: "The Tax Category to be assigned to the Product.",
+				Optional:            true,
+			},
+			"state_id": schema.StringAttribute{
+				Description:         "State to be assigned to the Product.",
+				MarkdownDescription: "State to be assigned to the Product.",
+				Optional:            true,
+			},
+			"publish": schema.BoolAttribute{
+				Description:         "If true, the Product is published immediately to the current projection. Default: false",
+				MarkdownDescription: "If `true`, the Product is published immediately to the current projection. Default: `false`",
+				Optional:            true,
+			},
+		},
+		Blocks: map[string]schema.Block{
```

Comment:
I think it would be better to follow the terraform suggestion here to not use blocks but instead nested attributes: https://developer.hashicorp.com/terraform/plugin/framework/handling-data/blocks. This is not something we have been doing consistently in the current code base, but is something I want to change to at some point (in a newer version of the provider). Would it be possible to make that change?

Reply:
Initially, I tried this one, but I encountered an error. I assumed that it may be due to the current "semi-migrated" state, but I could be mistaken.

```
2023/11/15 15:46:05 error retrieving schema for *proto5server.Server:
	Attribute: 
	Summary: Error converting resource schema
	Detail: The schema for the resource "commercetools_product" couldn't be converted into a usable type. This is always a problem with the provider. Please report the following to the provider developer:
AttributeName("example_attribute"): protocol version 5 cannot have Attributes set
```